qbraid/runtime/azure/provider.py

```diff
@@ -0,0 +1,494 @@
+            provider=provider,
+            backend=self.backends[backend],
+            num_qubits=qubits,
+            num_shots=100,
+            total_count=100,
+            output_data_format=self.formats[provider][1],
+        )
+
+        job_approval = self._helpers.submit_job(
+            job_id,
+            job_payload,
+            job_headers,
+            self.location_name,
+            self.subscription_id,
+            self.resource_group_name,
+            self.workspace_name,
+            self.session,
+        )
+
+        self.jobs[job_id] = [job_headers, container_name]
+
+        return job_approval
+
+    def get_job(self, job_id: str) -> dict[str, Any]:
+        """Get a specific Azure Quantum job."""
+
+        job = job_id
+        header = self.jobs[job_id][0]
+        url = (
+            f"https://{self.location_name}.quantum.azure.com/subscriptions/{self.subscription_id}"
+            f"/resourceGroups/{self.resource_group_name}/providers/Microsoft.Quantum/workspaces/"
+            f"{self.workspace_name}/jobs/{job}?api-version=2022-09-12-preview"
+        )
+
+        response = self.session.get(url, headers=header)
+        return response.json()
+
+    def get_job_data(self, job_id: str) -> dict[str, Any]:
+        """Get the data from a specific Azure Quantum job."""
+
+        data = self._helpers.get_output_data(f"{self.jobs[job_id][1]}", self.connection_string)
+        return data
+
+    def cancel_job(self, job_id: str):
+        """Cancel a specific Azure Quantum job."""
+        job = job_id
+        header = self.jobs[job_id][0]
+        url = (
+            f"https://{self.location_name}.quantum.azure.com/subscriptions/"
+            f"{self.subscription_id}/resourceGroups/{self.resource_group_name}/"
+            f"providers/Microsoft.Quantum/workspaces/{self.workspace_name}/jobs/"
+            f"{job}?api-version=2022-09-12-preview"
+        )
+
+        response = self.session.delete(url, headers=header)
+        print("Job has been cancelled.")
+        return str(response)
+
+
+class AzureQuantumProvider(QuantumProvider):
+    """Azure provider class."""
+
+    def __init__(
+        self,
+        client_id: str,
+        client_secret: str,
+        tenant_id: str,
+        location_name: str,
+        subscription_id: str,
+        resource_group_name: str,
+        workspace_name: str,
+        storage_account: str,
+        connection_string: str,
+    ):  # pylint: disable= too-many-arguments
+        super().__init__()
+        self.session = AzureSession(
+            client_id=client_id,
+            client_secret=client_secret,
+            tenant_id=tenant_id,
+            location_name=location_name,
+            subscription_id=subscription_id,
+            resource_group_name=resource_group_name,
+            workspace_name=workspace_name,
+            storage_account=storage_account,
+            connection_string=connection_string,
+        )
```

Comment:
Use `**kwargs` and pass directly through or require `AzureSession` as argument to avoid redundancy